sidecar/pkg/validator/validator.go

```diff
@@ -0,0 +1,140 @@
+// This file is part of MinIO Operator
+// Copyright (c) 2023 MinIO, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package validator
+
+import (
+	"bufio"
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	miniov2 "github.com/minio/operator/pkg/apis/minio.min.io/v2"
+	clientset "github.com/minio/operator/pkg/client/clientset/versioned"
+	"github.com/minio/operator/pkg/resources/statefulsets"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/klog/v2"
+)
+
+// Validate checks the configuration on the seeded configuration and issues a valid one for MinIO to
+// start, however if root credentials are missing, it will exit with error
+func Validate(tenantName string) {
+	rootUserFound, rootPwdFound, fileContents, err := ReadTmpConfig()
+	if err != nil {
+		panic(err)
+	}
+
+	namespace := miniov2.GetNSFromFile()
+
+	cfg, err := rest.InClusterConfig()
+	// If config is passed as a flag use that instead
+	//if kubeconfig != "" {
+	//	cfg, err = clientcmd.BuildConfigFromFlags(masterURL, kubeconfig)
+	//}
```

Comment:
needed?

Reply:
I guess this is useful to uncomment while debugging.